kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/ResponseHeaderAndApiMessage.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.Objects;
+
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ApiMessage;
+
+/**
+ * Tuple encapsulating a response message and its header.
+ * @param header message header
+ * @param message message
+ * @param <M> The type of the message
+ */
+public record ResponseHeaderAndApiMessage<M extends ApiMessage>(ResponseHeaderData header, M message) {
```

Comment:
`ResponseHeaderAndApiMessage` is a bit of a mouth full  and just repeats the type system but doesn't really tell me anything about what it represents or what it might be used for so how about? `ResponseEnvelope` which gives it a shorter and more usefully descriptive name. 

We can debate offline the merits or otherwise of having separate `RequestEnvelope` and `ResponseEnvelop` types...

Reply:
Good point.

The Kafka Protocol has the concept of Envelope (which is used broker to broker rather than client to broker).   Maybe best to steer clear. How about ResponseWrapper or ResponseContainer?

We don't need a Request* analogue for it.